### PR TITLE
Docker fix

### PR DESCRIPTION
**Pull request by jrottenberg**
Maybe some hope for #946

It's better than the first version, fresh docker-compose up should give you a working newsblur environment.

### CODE REVIEW

**Comment by samuelclay** [comment]
I would love to merge this but can you undo all the spacing changes? I just want the real diff. It looks like your text editor cleaned everything up. That would be fine, but I would want that as a separate PR for sure.

**Comment by jrottenberg** [comment]
np, I'll strip the formatting.

**Comment by jrottenberg** [comment]
ok rebased master and cleaned up the commits, the intent is more clear, let me know if anything is not explicit.

File: utils/feed_fetcher.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 from django.utils.encoding import smart_unicode
 from utils import json_functions as json
 from celery.exceptions import SoftTimeLimitExceeded
-from vendor import tweepy
```
Comment (samuelclay) [inline utils/feed_fetcher.py]: Can you undo these two? Tweepy is explicitly vendorized because of some changes that I needed to make.

Comment (samuelclay) [inline utils/feed_fetcher.py]: Aww geez, you're absolutely right. Thanks for catching this!
screen shot 2016-09-16 at sep 16 2 39 11 pm

**Comment by remram44** [comment]
I'm confused -- is the task server still missing from docker-compose?

**Comment by jrottenberg** [comment]
Not sure to follow you, the main service is labelled **newsblur** listening on port 8000. For the newsblur service it just starts the UI (gunicorn), I don't start any worker, it's doable but at the moment it's not clear what workers (see config/supervisor_*) are mandatory to be able to have a service running. I've defaulted to the simple webservice.

**Comment by remram44** [comment]
This is what I'm getting:
screenshot

I've added some celery workers but somehow they don't update the feeds? Worker log is full of:
```
celery_1         | [2016-10-01 02:07:18,138: INFO/MainProcess] Received task: task-feeds[af60f3d2-e960-46ba-93d2-e399bf351c48]
celery_1         | [Oct 01 02:07:18]  ---> No feeds to queue! Exiting...
```

I'm not sure how all the queues & tasks work, and the difference between feeds that are "tasked"/"queued"/"scheduled". My changes are here: https://github.com/remram44/NewsBlur/compare/master...docker-worker

**Comment by remram44** [comment]
Actually it seems like it's updating now... It looks like some task needs to go over the feeds and schedule them for updating? not sure what's this about.

**Comment by remram44** [comment]
Continuing my investigation, the feeds I upload as OPML don't seem to update until the `task-broken-feeds` happens and schedules them. This looks like a bug in either the timing of the task (only happens once every 6 hours) or the OPML importer (why are they added in a state that won't get them refreshed until the task fixes it?)

**Comment by samuelclay** [comment]
I answered this a bit in #946:

> So the fetching that's happening there is when the client sees that feeds aren't being updated and updates them itself (on the app servers). It essential,lly Insta-fetches the feeds since they aren't being updated in the background.

> You need to run the celery workers. I would just have them run every single queue and not split them up. If you omit the queue name, celery will just handle them all.

> You also need to setup celery's celery beat, which performs scheduled tasks. Those tasks are what fill the feed queue and then send the stale feeds out to the task queues.

**Comment by remram44** [comment]
I eventually figured that out, I'm just surprised that `broken-feeds` needs to go over them for them to start updating after an OPML import...?

I might open a pull request at some point, though realistically paying for your hosted service might be a better option, since this is too heavy for self-hosting on my home server 😅

**Comment by samuelclay** [comment]
Yeah, it is a lot of work to run. It's doing a ton of things, and that's assuming you've got search, socket.io/node, and all the queues set up. I'm not sure why the feeds didn't have next_scheduled_update times filled out correctly (that's what broken-feeds checks for), but that would explain why there's always a handful of feeds that fall through.

**Pull request merged**